File: src/main/java/eotree/EOBytesData.java
package eotree;

import java.util.List;
import java.util.stream.Collectors;

/**
 * EBNF representation:
 * <code>
 * byte { '-' byte }
 * </code>
 *
 */
public class EOBytesData extends EOData {
    public EOBytesData(List<EOByte> bytes) {
        this.bytes = bytes;
    }

    List<EOByte> bytes;

    @Override
    public String generateEO(int indent) {
        return bytes.stream()
                .map(b -> b.generateEO(indent))
                .collect(Collectors.joining("-"));
    }
}

File: src/main/java/eotree/data/EOBytesData.kt
package eotree.data

import java.util.*
import java.util.stream.Collectors

/**
 * EBNF representation:
 * `
 * byte { '-' byte }
` *
 */
class EOBytesData(var bytes: List<EOByte>) : EOData() {
    constructor(vararg bytes: EOByte) : this(Arrays.stream<EOByte>(bytes).collect(Collectors.toList<EOByte>())) {}

    override fun generateEO(indent: Int): String {
        return bytes.stream()
            .map { b: EOByte -> b.generateEO(indent) }
            .collect(Collectors.joining("-"))
    }
}